airflow/executors/kubernetes_executor.py

```diff
@@ -490,7 +490,13 @@ def execute_async(
     ) -> None:
         """Executes task asynchronously"""
         self.log.info('Add task %s with command %s with executor_config %s', key, command, executor_config)
-        kube_executor_config = PodGenerator.from_obj(executor_config)
+        try:
+            kube_executor_config = PodGenerator.from_obj(executor_config)
+        except Exception:  # pylint: disable=broad-except
+            self.log.error("Invalid executor_config for %s", key)
+            self.change_state(key=key, state=State.FAILED, info="Invalid executor_config passed")
```

Comment:
maybe can use the existing `self.fail()` method?

Reply:
I want to remove it from self.executor.running too which is done by self.change_state

Reply:
Updated